File: backend/app/schemas/movie.py
from pydantic import BaseModel, ConfigDict


class Movie(BaseModel):
    model_config = ConfigDict(extra="ignore")

    id: int
    title: str
    overview: str = ""
    poster_path: str | None = None
    backdrop_path: str | None = None
    poster_url: str | None = None
    backdrop_url: str | None = None
    release_date: str | None = None
    vote_average: float = 0
    popularity: float = 0
    original_language: str | None = None


class MovieListResponse(BaseModel):
    page: int = 1
    total_pages: int = 1
    total_results: int = 0
    results: list[Movie]

File: src/types/movie.ts
export type Movie = {
    id: number;
    title: string;
    overview?: string;
    poster_path: string | null;
    backdrop_path?: string | null;
    poster_url?: string | null;
    backdrop_url?: string | null;
    release_date?: string | null;
    vote_average?: number;
    popularity?: number;
    original_language?: string | null;
};

export type MovieListResponse = {
    page: number;
    total_pages: number;
    total_results: number;
    results: Movie[];
};
